lib/controller/StudyRequestController.js

```diff
@@ -231,4 +236,214 @@ StudyRequestController.push({
+      userSubject: subject,
+      ...request.payload,
+    });
+    return studyRequestComment;
+  },
+});
+
+/**
+ * Get comments for the given study request.
+ *
+ * @memberof StudyRequestController
+ * @name getStudyRequestComments
+ * @type {Hapi.ServerRoute}
+ */
+StudyRequestController.push({
+  method: 'GET',
+  path: '/requests/study/{studyRequestId}/comments',
+  options: {
+    response: {
+      schema: Joi.array.items(StudyRequestComment.read),
+    },
+    validate: {
+      params: {
+        studyRequestId: Joi.number().integer().positive().required(),
+      },
+      payload: {
+        ...StudyRequestComment.create,
+        // TODO: remove when we have RBAC
+        isSupervisor: Joi.boolean().default(false),
+      },
+    },
+  },
+  handler: async (request) => {
+    const { subject } = request.auth.credentials;
+    const { studyRequestId } = request.params;
+    const { isSupervisor } = request.query;
+    const studyRequest = await StudyRequestDAO.byId(studyRequestId);
+    if (studyRequest === null) {
+      return Boom.notFound(`no study request found with ID ${studyRequestId}`);
+    }
+    if (studyRequest.userSubject !== subject && !isSupervisor) {
+      return Boom.forbidden('cannot view comments for study request owned by another user');
+    }
+    return StudyRequestCommentDAO.byStudyRequest(studyRequest);
+  },
+});
+
+/**
+ * Update the given comment on the given study request.
+ *
+ * The request body should contain the comment.
+ *
+ * The ID of the request URI should match the ID of the comment in the body.
+ *
+ * @memberof StudyRequestController
+ * @name putStudyRequestComment
+ * @type {Hapi.ServerRoute}
+ */
+StudyRequestController.push({
+  method: 'PUT',
+  path: '/requests/study/{studyRequestId}/comments/{commentId}',
+  options: {
+    response: {
+      schema: StudyRequest.read,
+    },
+    validate: {
+      params: {
+        commentId: Joi.number().integer().positive().required(),
+        studyRequestId: Joi.number().integer().positive().required(),
+      },
+      payload: {
+        ...StudyRequest.update,
+        // TODO: remove when we have RBAC
+        isSupervisor: Joi.boolean().default(false),
+      },
+    },
+  },
+  handler: async (request) => {
+    const user = request.auth.credentials;
+    const { commentId, studyRequestId } = request.params;
+    const { isSupervisor, ...commentNew } = request.payload;
+
+    const studyRequest = await StudyRequestDAO.byId(studyRequestId);
+    if (studyRequest === null) {
+      return Boom.notFound(`no study request found with ID ${studyRequestId}`);
+    }
+    const commentOld = await StudyRequestCommentDAO.byId(commentId);
+    if (commentOld.studyRequestId !== studyRequest.id) {
+      return Boom.badRequest(
+        `no comment with ID ${commentId} for study request ${studyRequestId}`,
+      );
+    }
+
+    if (commentOld.id !== commentNew.id) {
```

Comment:
it feels kind of weird to do all this work to essentially update one field, but maybe this is OK if we ever need to add more fields to `study_request_comments`?